javacord-api/src/main/java/org/javacord/api/entity/server/scheduledevent/ServerScheduledEventBuilder.java

```diff
@@ -0,0 +1,149 @@
+import org.javacord.api.util.internal.DelegateFactory;
+import java.io.File;
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This class is used to create server scheduled events.
+ * See <a href="https://discord.com/developers/docs/resources/guild-scheduled-event#create-guild-scheduled-event">discord documentation</a> for required fields to create events.
+ */
+public class ServerScheduledEventBuilder {
+
+    /**
+     * The server scheduled event delegate used by this instance.
+     */
+    private final ServerScheduledEventBuilderDelegate delegate;
+
+    /**
+     * Creates a new server scheduled event builder for the given server.
+     *
+     * @param server The server for which the server scheduled event should be created.
+     */
+    public ServerScheduledEventBuilder(Server server) {
+        delegate = DelegateFactory.createServerScheduledEventBuilderDelegate(server);
+    }
+
+    /**
+     * Sets the reason for the creation. This reason will be visible in the audit log entry(s).
+     *
+     * @param reason The reason for this update.
+     * @return The current instance in order to chain call methods.
+     */
+    public ServerScheduledEventBuilder setAuditLogReason(String reason) {
+        delegate.setAuditLogReason(reason);
+        return this;
+    }
+
+    /**
+     * Sets the name of the server scheduled event.
+     *
+     * @param name The name of the server scheduled event.
+     * @return The current instance in order to chain call methods.
+     */
+    public ServerScheduledEventBuilder setName(String name) {
+        delegate.setName(name);
+        return this;
+    }
+
+    /**
+     * Sets the description of the server scheduled event.
+     *
+     * @param description The description of the server scheduled event.
+     * @return The current instance in order to chain call methods.
+     */
+    public ServerScheduledEventBuilder setDescription(String description) {
+        delegate.setDescription(description);
+        return this;
+    }
+
+    /**
+     * Sets the scheduled start time of the server scheduled event.
+     *
+     * @param scheduledStartTime The scheduled start time of the server scheduled event.
+     * @return The current instance in order to chain call methods.
+     */
+    public ServerScheduledEventBuilder setScheduledStartTime(Instant scheduledStartTime) {
+        delegate.setScheduledStartTime(scheduledStartTime);
+        return this;
+    }
+
+    /**
+     * Sets the scheduled end time of the server scheduled event.
+     *
+     * @param scheduledEndTime The scheduled end time of the server scheduled event.
+     * @return The current instance in order to chain call methods.
+     */
+    public ServerScheduledEventBuilder setScheduledEndTime(Instant scheduledEndTime) {
+        delegate.setScheduledEndTime(scheduledEndTime);
+        return this;
+    }
+
```

Comment:
```suggestion

    /**
     * Sets the scheduled start and end time of the server scheduled event.
     *
     * @param scheduledStartTime The scheduled start time of the server scheduled event.
     * @param duration The intended duration to calculate the scheduled end time.
     * @return The current instance in order to chain call methods.
     */
    public ServerScheduledEventBuilder setScheduledStartAndEndTimes(Instant scheduledStartTime, Duration duration) {
        return setScheduledStartTime(scheduledStartTime).setScheduledEndTime(scheduledStartTime.plus(duration));
    }
```
Another convenience method that I reckon will address a common use case. Needs another import to fully work.

Maybe on the updater too?